### PR TITLE
Upgade to eventhub 2.2.1

### PR DESCRIPTION
**Pull request by jason-bragg**

### CODE REVIEW

File: src/Azure/Orleans.Streaming.EventHubs/Providers/Streams/EventHub/EventHubConstants.cs [hunk 1]
```diff
@@ -2,7 +2,6 @@ namespace Orleans.ServiceBus.Providers
 {
     internal class EventHubConstants
     {
-        public readonly static string StartOfStream =
-            Microsoft.Azure.EventHubs.PartitionReceiver.StartOfStream;
```
Comment (xiazen) [inline src/Azure/Orleans.Streaming.EventHubs/Providers/Streams/EventHub/EventHubConstants.cs]: why this change? there isn't something like `Microsoft.Azure.EventHubs.PartitionReceiver.StartOfStream` in new pkg anymore?

File: src/Azure/Orleans.Streaming.EventHubs/Providers/Streams/EventHub/EventHubAdapterFactory.cs [hunk 2]
```diff
@@ -96,7 +96,7 @@ public class EventHubAdapterFactory : IQueueAdapterFactory, IQueueAdapter, IQueu
         /// <summary>
         /// Factory to create a IEventHubReceiver
         /// </summary>
-        protected Func<EventHubPartitionSettings, string, ILogger, ITelemetryProducer, Task<IEventHubReceiver>> EventHubReceiverFactory;
```
Comment (xiazen) [inline src/Azure/Orleans.Streaming.EventHubs/Providers/Streams/EventHub/EventHubAdapterFactory.cs]: yeah I can see using API in the new evenhub pkg, a async Task isn't needed anymore. But this is a breaking change indeed. I don't have strong opinion on this, since most people won't be affected. Just want to point that out.

File: src/Azure/Orleans.Streaming.EventHubs/Providers/Streams/EventHub/EventHubConstants.cs (hunk 1 above)
Comment (jason-bragg) [reply src/Azure/Orleans.Streaming.EventHubs/Providers/Streams/EventHub/EventHubConstants.cs]: No, they don't use '-1' as a signal offset for the start of the partition, but we still need it to not break backwards compatibility with checkpoint entities that may have been written with it.
